modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Engine.scala

```diff
@@ -410,6 +413,7 @@ object Engine {
 
     def empty[D](userData: D): State[D] = State(userData, Map.empty)
 
+    implicit def equal[D]: Equal[State[D]] = Equal.equalA
```

Comment:
I think I would require `Equal[D]` here and contramap from `Equal[(D, Map[Sequence.Id, Sequence.State])]` otherwise it's not going to be lawful in general.

Reply:
```scala
implicit def stateEqual[D: Equal]: Equal[State[D]] =
  Equal[(D, Map[Sequence.Id, Sequence.State])]
    .contramap(s => (s.userData, s. sequences))
```

Reply:
ok, I couldn't do that as i don't have all of them defined. I'll add those.
On the cats port I'm doing `discipline` testing. Could this problem have been detected that way?